src/rno-g-acq.c

```diff
@@ -567,17 +573,110 @@ int flower_initial_setup()
     flower_set_trigger_enables(flower,trig_enables);
     flower_equalize(flower, target,flower_codes,FLOWER_EQUALIZE_VERBOSE); 
   }
-  
+
+
+  if ( cfg.lt.waveforms.length > 0 && (cfg.lt.waveforms.at_start.enable || cfg.lt.waveforms.at_finish.enable))
+  {
+    flower_waveforms_data = calloc(RNO_G_NUM_LT_CHANNELS, cfg.lt.waveforms.length);
+    for (int i = 0; i < RNO_G_NUM_LT_CHANNELS; i++) 
+    {
+      flower_waveforms[i] = flower_waveforms_data + i * cfg.lt.waveforms.length;
+    }
+  }
 
   flower_set_thresholds(flower,  ds->lt_trigger_thresholds, ds->lt_servo_thresholds, 0xf); 
   //then the rest of the configuration; 
   flower_configure(); 
 
 
+
   return 0; 
 }
 
 
+//right now this can only run in the main thread before and after data taking!!!
+int flower_take_waveform(gzFile of, int LEN, int force, int iev, struct timespec * deadline)
```

Comment:
why capital `LEN`?

Reply:
because I didn't change it from the original flower-wave code where it was a constant :).  Refactoring so that the length is stored when the memory is allocated (technically the config can be reread during the run, but we don't want this to change).